block/manager.go

```diff
@@ -604,14 +631,6 @@ func (m *Manager) publishBlockInternal(ctx context.Context) error {
 		return SaveBlockError{err}
 	}
 
-	// Commit the new state and block which writes to disk on the proxy app
-	appHash, err := m.execCommit(ctx, newState, header, data)
```

Comment:
may be a dumb question, but now that the publication is in a seperaate go routine, does this mean SetFinal must not depend on any other app state / flow? Is that noted somewhere for future implementors of the exec interface?